PusherMainView/PusherMainView/PusherInteractor.swift

```diff
@@ -140,12 +143,75 @@ extension PusherInteractor: PusherInteracting {
                                 }
         })
     }
+}
+
+extension PusherInteractor: PusherInteracting {
+    func subscribe(_ pusherInteractable: PusherInteractable) {
+        subscribers.append(pusherInteractable)
+    }
     
-    func cancelSelectingAuthToken() {
-        delegate?.didCancelSelectingAuthToken()
+    func unsubscribe(_ pusherInteractable: PusherInteractable) {
+        guard let index = subscribers.firstIndex(where: { pusherInteractable == $0 }) else {
+            return
+        }
+        subscribers.remove(at: index)
     }
     
-    func selectDevice(_ device: APNSServiceDevice) {
-        delegate?.didSelectDevicetoken(device.token, appBundleID: device.appID)
+    func dispatch(actionType: ActionType) {
```

Comment:
In redux I think the state is changed in the reducer functions.

Reply:
Yes it is true. For this PR, I let state changes here.